File: CS-Examples/04_RowsColumns/RemoveRowBasedOnKeyword.cs
using Spire.Xls;
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;

namespace RemoveRowBasedOnKeyword
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void btnRun_Click(object sender, EventArgs e)
        {
            // Create a new workbook object
            Workbook workbook = new Workbook();

            // Load an existing file from disk
            workbook.LoadFromFile(@"..\..\..\..\..\..\Data\WorkbookToHTML.xlsx");

            // Get the first worksheet in the workbook
            Worksheet sheet = workbook.Worksheets[0];

            // Find the string "Address" in the worksheet
            CellRange cr = sheet.FindString("Address", false, false);

            // Delete the row that includes the found string
            sheet.DeleteRow(cr.Row);

            // Save the modified workbook to a new file
            workbook.SaveToFile("RemoveRowBasedOnKeyword.xlsx", ExcelVersion.Version2010);

            // Dispose of the workbook object to release resources
            workbook.Dispose();

            //View the document
            FileViewer("RemoveRowBasedOnKeyword.xlsx");
        }

        private void FileViewer(string fileName)
        {
            try
            {
                System.Diagnostics.Process.Start(fileName);
            }
            catch { }
        }

        private void btnClose_Click(object sender, EventArgs e)
        {
            Close();
        }
    }
}

File: CS-Examples/04_RowsColumns/RemoveRowBasedOnKeyword.vb
Imports Spire.Xls
Imports System.ComponentModel
Imports System.Text

Namespace RemoveRowBasedOnKeyword
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub btnRun_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnRun.Click
            'Creates a new instance of the Workbook class.
            Dim workbook As New Workbook()
            'Loads the Excel document from the specified path.
            workbook.LoadFromFile("..\..\..\..\..\..\Data\WorkbookToHTML.xlsx")
            'Retrieves the first worksheet from the workbook.
            Dim sheet As Worksheet = workbook.Worksheets(0)

            'Find the string "Address" in the worksheet.
            Dim cr As CellRange = sheet.FindString("Address", False, False)

            'Delete the row which includes the found string.
            sheet.DeleteRow(cr.Row)

            'Save the modified workbook to a file named "RemoveRowBasedOnKeyword.xlsx" in Excel 2010 format.
            workbook.SaveToFile("RemoveRowBasedOnKeyword.xlsx", ExcelVersion.Version2010)
            ' Release the resources used by the workbook
            workbook.Dispose()

            'View the document
            FileViewer("RemoveRowBasedOnKeyword.xlsx")
		End Sub

		Private Sub FileViewer(ByVal fileName As String)
			Try
				Process.Start(fileName)
			Catch
			End Try
		End Sub

		Private Sub btnClose_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnClose.Click
			Close()
		End Sub
	End Class
End Namespace
